File: Bomba.java
public class Bomba {
    private final Combustivel combustivel;
    private final double velocidadeAbastecimento;
    private double totalAbastecido = 0;
    public static double totalGasolina = 0;
    public static double totalEtanol = 0;

    public Bomba(Combustivel combustivel, double velocidadeAbastecimento) {
        this.combustivel = combustivel;
        this.velocidadeAbastecimento = velocidadeAbastecimento;
    }

    public double getVelocidadeAbastecimento() {
        return velocidadeAbastecimento;
    }
    public Combustivel getCombustivel() {
        return combustivel;
    }
    public double getTotalAbastecido() {
        return totalAbastecido;
    }

    /**
     * Abastece um veículo e atualiza o total de combustível abastecido.
     * <p>
     * Este método abastece um veículo com base na capacidade do tanque do veículo.
     * Ele assume que o tanque do veículo está vazio. O total de combustível abastecido pela bomba é atualizado.
     * Além disso, o total de gasolina ou etanol abastecido é atualizado com base no tipo de combustível da bomba.
     * <p>
     * @param veiculo O veículo a ser abastecido.
     * @return A quantidade de combustível (em litros) com que o veículo foi abastecido.
     */
    public double abastecer(Veiculo veiculo) {
        double litrosAbastecidos = Double.parseDouble(veiculo.getCapacidadeTanque()); // Supondo que o tanque esteja vazio
        totalAbastecido += litrosAbastecidos;
        if (combustivel.getTipoCombustivel() == TipoCombustivel.GASOLINA)
            totalGasolina += litrosAbastecidos;
        else if (combustivel.getTipoCombustivel() == TipoCombustivel.ETANOL)
            totalEtanol += litrosAbastecidos;
        return litrosAbastecidos;
    }

    @Override
    public String toString() {
        return "Bomba {\n" +
                "\tcombustivel = " + combustivel + ",\n" +
                "\tvelocidadeAbastecimento = " + velocidadeAbastecimento + "\n" +
                '}';
    }
}

File: Python/Bomba.py
from Combustivel import TipoCombustivel


class Bomba:
    total_gasolina = 0
    total_etanol = 0

    def __init__(self, combustivel, velocidade_abastecimento):
        self.combustivel = combustivel
        self.velocidade_abastecimento = velocidade_abastecimento
        self.total_abastecido = 0

    def get_velocidade_abastecimento(self):
        return self.velocidade_abastecimento

    def get_combustivel(self):
        return self.combustivel

    def get_total_abastecido(self):
        return self.total_abastecido

    def abastecer(self, veiculo):
        litros_abastecidos = float(veiculo.get_capacidade_tanque())  # Supondo que o tanque esteja vazio
        self.total_abastecido += litros_abastecidos
        if self.combustivel.get_tipo_combustivel() == TipoCombustivel.GASOLINA:
            Bomba.total_gasolina += litros_abastecidos
        elif self.combustivel.get_tipo_combustivel() == TipoCombustivel.ETANOL:
            Bomba.total_etanol += litros_abastecidos
        return litros_abastecidos

    def __str__(self):
        return f"Bomba {{\n\tcombustivel = {self.combustivel},\n\tvelocidade_abastecimento = {self.velocidade_abastecimento}\n}}"
